### PR TITLE
eigen3 is now a required dependency.  matlab is an optional dependency.

### LINKED ISSUE #974: Eigen as a required dependency?
I'd like to propose making Eigen a required dependency. My arguments:
* As we're moving more stuff to C++, a larger percentage of our codebase will be unusable without Eigen.
* It's a hassle to remember to use if (eigen3_FOUND) everywhere.
* Why would any system we care about not have Eigen?

What do you think?

### PR DESCRIPTION
**Pull request by RussTedrake**
resolves #974

### CODE REVIEW

File: solvers/qpSpline/CMakeLists.txt [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
+pods_install_pkg_config_file(drakeSplineGeneration
+  LIBS -ldrakeSplineGeneration
+  REQUIRES
+  VERSION 0.0.2)
+
+if (MATLAB_FOUND)
   add_mex(twoWaypointCubicSplinemex twoWaypointCubicSplinemex.cpp)
-  include_directories (${CMAKE_SOURCE_DIR}/util)
-  include_directories (${CMAKE_SOURCE_DIR}/solvers/polynomial)
-  include_directories (${CMAKE_SOURCE_DIR}/systems/trajectories)
```
Comment (tkoolen) [inline solvers/qpSpline/CMakeLists.txt]: Why do we not need these?

File: util/CMakeLists.txt [hunk 2]
```diff
@@ -101,4 +103,4 @@ file(APPEND ${CMAKE_BINARY_DIR}/bin/drake_debug_mex.sh
 install(FILES ${CMAKE_BINARY_DIR}/bin/drake_debug_mex.sh
         DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
         PERMISSIONS OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
-endif(NOT WIN32)
+endif(0) #NOT WIN32)
```
Comment (tkoolen) [inline util/CMakeLists.txt]: No more warning :-) thanks.

**Comment by tkoolen** [comment]
Wow.

File: solvers/qpSpline/CMakeLists.txt (hunk 1 above)
Comment (RussTedrake) [inline solvers/qpSpline/CMakeLists.txt]: Moved them above.  In fact i don't know how the drakeSplineGeneration was working without them

**Pull request merged**